ops/model.py

```diff
@@ -3347,7 +3355,9 @@ def _run(
         # TODO(benhoyt): all the "type: ignore"s below kinda suck, but I've
         #                been fighting with Pyright for half an hour now...
         try:
-            result = subprocess.run(args, **kwargs)  # type: ignore
+            with tracer.start_as_current_span('ops.model::subprocess.run') as span:
+                span.set_attribute('argv', args)
+                result = subprocess.run(args, **kwargs)  # type: ignore
```

Comment:
I think this is fine, with two exceptions:

* `action-set`: this definitely contains sensitive data sometimes (like an "add super user" action that returns credentials). Annoyingly, there's no `--file` option according to the documentation, so we can't do that like we do with relation and secret data. Maybe there needs to be a "sensitive args" flag or something to skip adding it in this case? Or just hard-code an exception for `action-set`?
* `juju-log`: it feels a bit excessive to include all of the logging inside of the trace in this way. I think we would want to ignore the args in this case.

Reply:
juju-log should be excluded via `self.drop` in the `JujuLogHandler`.

I'll keep this open to confirm that.